server/routes/auth.ts

```diff
@@ -133,6 +138,126 @@ authRoutes.post('/login', async (req, res, next) => {
   }
 });
 
+authRoutes.post('/jellyfin', async (req, res, next) => {
+  const settings = getSettings();
+  const userRepository = getRepository(User);
+  const body = req.body as {
+    username?: string;
+    password?: string;
+    hostname?: string;
+  };
+
+  //Make sure jellyfin login is enabled, but only if jellyfin is not already configured
+  if (
+    settings.main.mediaServerType != 'JELLYFIN' &&
+    settings.jellyfin.hostname != ''
+  ) {
+    return res.status(500).json({ error: 'Jellyfin login is disabled' });
+  } else if (!body.username || !body.password) {
+    return res
+      .status(500)
+      .json({ error: 'You must provide an username and a password' });
+  } else if (settings.jellyfin.hostname != '' && body.hostname) {
+    return res
+      .status(500)
+      .json({ error: 'Jellyfin hostname already configured' });
+  } else if (settings.jellyfin.hostname == '' && !body.hostname) {
+    return res.status(500).json({ error: 'No hostname provided.' });
+  }
```

Comment:
Why do we need to provide the hostname here instead of just using what ever is set in settings, and sending 500 only if no hostname is set in settings?

Reply:
I have the hostname provided here due to how Jellyfin works. I decided to allow the setup flow to send the `hostname` attr, and it will not save until a successful authorization. Then, the hostname value is locked and cannot be sent in any future login requests. This is so a user could not send a login request with a hostname other than what the server is configured for.